packages/astro/src/core/dev/dev.ts

```diff
@@ -32,7 +34,26 @@ export interface DevServer {
 export default async function dev(
 	settings: AstroSettings,
 	options: DevOptions
-): Promise<DevServer> {
+): Promise<DevServer | undefined> {
+	if (options.flags?.help || options.flags?.h) {
+		printHelp({
+			commandName: 'astro dev',
+			usage: '[...flags]',
+			tables: {
+				Flags: [
+					['--port', `Specifies which port to run on. Defaults to 3000.`],
+					['--host', `listen on all addresses, including LAN and public addresses.`],
+					['--host <custom-address>', `expose on a network IP address at <custom-address>`],
+					['--help (-h)', 'See all available flags.'],
```

Comment:
```suggestion
					['--port', `Specifies which port to run on. Defaults to 3000.`],
					['--host', `listen on all addresses, including LAN and public addresses.`],
					['--host <custom-address>', `expose on a network IP address at <custom-address>`],
					['--help (-h)', 'See all available flags.'],
```
Since these are messages to print, I'll just point out FYI that these are not consistent. Some begin with capital letters, some end with periods. The starting verb tense isn't consistent.  :)

If you wanted consistency, you could choose the "command" tense of each (Specify, Listen, Expose, See) or the "descriptive" tense of what the flag does (Specifies, Listens, Exposes, Displays/Prints/Lists) but See is the tricky one to replace here, and it's probably not horrible if that one just stays as "See". But, ideally the top 3 should at least be consistent, to me.

Up to you! Just noting!

Reply:
Would be pretty cool to generate the Docs descriptions from the help commands, though those are definitely different contexts